### PR TITLE
Remove dead code in tasks. Translation strings that are not yet trans…

### PR DESCRIPTION
**Pull request by wawax**
- Clean unused methods
- Use constants for tasks related states
- Translate hardcoded strings in tasks logs

### CODE REVIEW

File: inc/communicationnetworkdiscovery.class.php [hunk 1]
```diff
@@ -103,45 +103,32 @@ function import($p_DEVICEID, $a_CONTENT, $arrayinventory) {
       }
 
       if ($pfTaskjobstate->getFromDB($a_CONTENT['PROCESSNUMBER'])) {
-         if ($pfTaskjobstate->fields['state'] != "3") {
+         if ($pfTaskjobstate->fields['state'] != PluginFusioninventoryTaskjobstate::FINISHED) {
             $pfImportExport = new PluginFusioninventorySnmpmodelImportExport();
             $errors .= $pfImportExport->import_netdiscovery($a_CONTENT, $p_DEVICEID);
             if (isset($a_CONTENT['AGENT']['END'])) {
-               if ((isset($a_CONTENT['DICO'])) AND ($a_CONTENT['DICO'] == "REQUEST")) {
```
Comment (trasher) [inline inc/communicationnetworkdiscovery.class.php]: I really do not know how to be sure this is still used

File: inc/taskjob.class.php [hunk 2]
```diff
@@ -530,15 +447,9 @@ function dropdownAction($myname, $actiontype, $method, $actiontypeid, $value=0,
...
    function getAgents($module) {
-
-      $array = [];
-      $array[".1"] = " ".__('Auto managenement dynamic of agents', 'fusioninventory');
-
-      $array[".2"] = " ".__('Auto managenement dynamic of agents (same subnet)', 'fusioninventory');
-
+      $array         = [];
       $pfAgentmodule = new PluginFusioninventoryAgentmodule();
-      $array1 = $pfAgentmodule->getAgentsCanDo(strtoupper($module));
```
Comment (trasher) [inline inc/taskjob.class.php]: Hum... Maybe the original way was better for perfs

File: inc/communicationnetworkdiscovery.class.php (hunk 1 above)
Comment (wawax) [reply inc/communicationnetworkdiscovery.class.php]: it was an old behavior of the agent:it had an auto update mechanism for network discovery mapping file. If the agent had a outdated version, the server would send a message to stop the process. Next time, the server would send the new file to the agent and the process could continue.
If was complicated and frustrating for the user, it has been removed a long time ago.

Comment (trasher) [reply inc/communicationnetworkdiscovery.class.php]: OK, thank you for the explanations :)

File: inc/taskjob.class.php (hunk 2 above)
Comment (wawax) [reply inc/taskjob.class.php]: you're right I've reverted this modification

File: inc/taskjoblog.class.php [hunk 3]
```diff
@@ -899,40 +686,33 @@ function getDivState($state, $type='div') {
...
 
-         case 2:
+         case self::TASK_OK:
             return "<".$type." style='background-color: rgb(0, 255, 0);-moz-border-radius: 4px;".
                      "-webkit-border-radius: 4px;-o-border-radius: 4px;padding: 2px;' ".
                      "align='center' width='".$width."'>".
-                     "<strong>".__('Ok', 'fusioninventory')."</strong></".$type.">";
+                     "<strong>".__('OK')."</strong></".$type.">";
 
-         case 3:
```
Comment (trasher) [inline inc/taskjoblog.class.php]: I guess this is also something that is no longer used now?

File: inc/taskjob.class.php [hunk 4]
```diff
@@ -530,19 +447,18 @@ function dropdownAction($myname, $actiontype, $method, $actiontypeid, $value=0,
     * @return array [id integed agent id] => $name value agent name
     */
    function getAgents($module) {
-
-      $array = [];
-      $array[".1"] = " ".__('Auto managenement dynamic of agents', 'fusioninventory');
```
Comment (trasher) [inline inc/taskjob.class.php]: Another old thing not longer used?

Comment (wawax) [reply inc/taskjob.class.php]: yes, it was an old task feature that has been deprecated during the task rework

File: inc/taskjoblog.class.php (hunk 3 above)
Comment (wawax) [reply inc/taskjoblog.class.php]: yes, this taskjoblog state has never been used...

**Pull request merged**